sql-plugin/src/main/scala/com/nvidia/spark/rapids/profiler.scala

```diff
@@ -246,13 +273,18 @@ object ProfilerOnExecutor extends Logging {
       val (completedJobs, completedStages, allDone) =
         w.pluginCtx.ask(ProfileJobStageQueryMsg(jobs, stages))
```

Comment:
One problem with removing from activeStages when the last "interesting" task completes in a stage is that we won't ask the driver about this stage when we poll, and thus won't be informed when it completes to know when we can clear out the task counts for that stage.  One fix for that is something like this when we build the jobs and stages for querying the driver:
```scala
      val (jobs, stages) = synchronized {
        (activeJobs.toArray, (activeStages ++ stageTaskCount.keys).toArray)
      }
```

Reply:
thanks, done.